src/test/java/com/meta/cp4m/S3PreProcessorTest.java

```diff
@@ -0,0 +1,33 @@
+/*
+ *
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.meta.cp4m;
+
+import com.meta.cp4m.plugin.DummyPlugin;
+import com.meta.cp4m.message.WAMessage;
+import com.meta.cp4m.message.WAMessageHandler;
+import com.meta.cp4m.message.WAMessengerConfig;
+import com.meta.cp4m.store.NullStore;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class S3PreProcessorTest {
```

Comment:
I get that this will be a hard test to write for s3, for the sake of this detail in the description for this PR how you manually tested this and then in a later PR we also need a test for preprocessors in general